Error deploying mule helloworld example

I get a maven resolution error while downloading the mule container on the very latest master.

This is because the mule container version was very recently changed to [3.5.0](https://github.com/fabric8io/fabric8/blob/master/fabric/fabric8-karaf/src/main/resources/distro/fabric/import/fabric/profiles/containers/mule.ce.profile/io.fabric8.container.process.properties#L20).

This is because 3.5.0 has [not](http://repo1.maven.org/maven2/org/mule/distributions/mule-standalone/) been uploaded to maven central :-( .

I asked MuleSoft recently [when they were going to do this](http://forum.mulesoft.org/mulesoft/topics/when-will-mule-3-5ce-distrbution-be-uploaded-to-maven-central). They said they had deliberately stopped uploading distributions to central, but didn't explain why.

To make this work, we would have to reference the [mulesoft repository](https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/distributions/mule-standalone/3.5.0/) in the profile itself (ideally so the repository is only used for that profile).

I tried this once before, but was unable to get it to work. However the [fabric8 docs](http://fabric8.io/gitbook/mavenProxy.html) seem to suggest it is possible. So I had planned to tackle this sometime after I had 'tidied' the mule example. :-)

```
2014-09-27 20:26:25,191 | INFO  | pool-38-thread-1 | MavenProxyServletSupport         | ?                                   ? | 184 - io.fabric8.fabric-maven-proxy - 1.2.0.SNAPSHOT | Received request for maven artifact : org/mule/distributions/mule-standalone/3.5.0/mule-standalone-3.5.0.zip
2014-09-27 20:26:29,852 | WARN  | pool-38-thread-1 | MavenProxyServletSupport         | ?                                   ? | 184 - io.fabric8.fabric-maven-proxy - 1.2.0.SNAPSHOT | Could not find artifact : org.mule.distributions:mule-standalone:zip:3.5.0 due to org.sonatype.aether.resolution.ArtifactResolutionException: Could not find artifact org.mule.distributions:mule-standalone:zip:3.5.0 in central (http://repo1.maven.org/maven2)
org.sonatype.aether.resolution.ArtifactResolutionException: Could not find artifact org.mule.distributions:mule-standalone:zip:3.5.0 in central (http://repo1.maven.org/maven2)
    at org.sonatype.aether.impl.internal.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:541)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.impl.internal.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:220)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.impl.internal.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:197)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.impl.internal.DefaultRepositorySystem.resolveArtifact(DefaultRepositorySystem.java:323)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at io.fabric8.maven.impl.MavenProxyServletSupport.download(MavenProxyServletSupport.java:233)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at io.fabric8.maven.impl.MavenDownloadProxyServlet$ArtifactDownloadTask.call(MavenDownloadProxyServlet.java:154)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at io.fabric8.maven.impl.MavenDownloadProxyServlet$ArtifactDownloadTask.call(MavenDownloadProxyServlet.java:144)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)[:1.7.0_67]
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)[:1.7.0_67]
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)[:1.7.0_67]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)[:1.7.0_67]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)[:1.7.0_67]
    at java.lang.Thread.run(Thread.java:745)[:1.7.0_67]
Caused by: org.sonatype.aether.transfer.ArtifactNotFoundException: Could not find artifact org.mule.distributions:mule-standalone:zip:3.5.0 in central (http://repo1.maven.org/maven2)
    at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$4.wrap(WagonRepositoryConnector.java:945)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$4.wrap(WagonRepositoryConnector.java:940)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$GetTask.flush(WagonRepositoryConnector.java:695)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$GetTask.flush(WagonRepositoryConnector.java:689)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.connector.wagon.WagonRepositoryConnector.get(WagonRepositoryConnector.java:445)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    at org.sonatype.aether.impl.internal.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:460)[184:io.fabric8.fabric-maven-proxy:1.2.0.SNAPSHOT]
    ... 12 more
```

--- a/fabric/fabric8-karaf/src/main/resources/distro/fabric/import/fabric/profiles/containers/mule.ce.profile/io.fabric8.container.process.properties
+++ b/fabric/fabric8-karaf/src/main/resources/distro/fabric/import/fabric/profiles/containers/mule.ce.profile/io.fabric8.container.process.properties
@@ -17,7 +17,7 @@
 processName=Mule Community Edition
 
 # which version of Mule to use
-processVersion = 3.5.0
+processVersion = 3.4.0
 
 url=mvn\:org.mule.distributions/mule-standalone/${processVersion}/zip
 

